lib/jnpr/junos/factcache.py

```diff
@@ -0,0 +1,280 @@
+from abc import ABCMeta
+import collections
+import warnings
+
+import jnpr.junos.facts
+import jnpr.junos.exception
+
+
+class _FactCache(collections.MutableMapping):
```

Comment:
Yes. This is what I was trying to work around with metaclasses. However, that did not work as expected. I'm open to any suggestions on how to workaround this restriction. However, since direct type-testing is not recommended best practice in Python, and this class does support duck typing, I don't expect this to be a significant issue.